File: packages/rook_sdk_samsung_health/lib/src/data/mapper/rook_environment_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_samsung_health/src/data/proto/protos.pbenum.dart';

extension RookEnvironmentMappers on RookEnvironment {
  RookEnvironmentProto toProto() {
    return switch (this) {
      RookEnvironment.sandbox => RookEnvironmentProto.SANDBOX,
      _ => RookEnvironmentProto.PRODUCTION,
    };
  }
}

File: packages/rook_sdk_samsung_health/android/src/main/kotlin/io/tryrook/rook_sdk_samsung_health/mapper/RookEnvironmentMappers.kt
import io.tryrook.rook_sdk_samsung_health.proto.RookEnvironmentProto
import io.tryrook.sdk.samsung.domain.environment.SHEnvironment

fun RookEnvironmentProto.toSHEnvironment(): SHEnvironment {
    return if (this == RookEnvironmentProto.SANDBOX) {
        SHEnvironment.SANDBOX
    } else {
        SHEnvironment.PRODUCTION
    }
}
